viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]['title'] == title:
+            user_data['watched'].append(user_data['watchlist'][i])
+            user_data["watchlist"].remove(user_data["watchlist"][i])
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    '''Calculate avg rating of watched movies in user data'''
+    average = 0.0
+    rating_total = 0.0
+    
+    for i in range(len(user_data["watched"])):
+        rating_total += user_data["watched"][i]["rating"]
+    
+    if len(user_data['watched']) == 0:
+        average == 0.0
+    else:
+        average = rating_total / len(user_data["watched"])
+    
+    return average
+
+
+def get_most_watched_genre(user_data):
+    '''Returning the genre watched the most times in user_data'''
+    genres = []
+    genre_count = {}
+    
+    for i in range(len(user_data["watched"])):
+        genres.append(user_data['watched'][i]['genre'])
+
+    # create dict of genres and how many times they appear in genres list
+    for item in genres:
+        genre_count[item] = 0   # add item to genre_count dict with value of 0 to start
+        genre_count[item] += 1  # add 1 to genre count item for each time it appears in genre list
+
+    # get max of genre count in dict
+    if len(genre_count) > 0:
+        max_genre = max(genre_count, key=genre_count.get)       
+        return max_genre
+    else:
+        return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+from copy import copy
+def get_unique_watched(user_data):
+    '''Returning list of unique movies watched by user'''
+    # make copy of user_data watched to use as list to remove movies also watched by friends
+    unique_movies = copy(user_data["watched"])
+
+    for my_movie in user_data["watched"]: #access list in user watched
+        for friend in user_data["friends"]: #access friend lists in user friends
+            for friend_movie in friend["watched"]: #access movie list in each friend watched
+                if my_movie["title"] == friend_movie["title"] and my_movie in unique_movies:
+                    unique_movies.remove(my_movie)
+
+    return unique_movies
+    
+ 
+def get_friends_unique_watched(user_data):
+    '''Returning list of unique movies watched by friends and not user'''
+    friend_movies = []
+
+    #create combined friend movie list
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friend_movies:
+                friend_movies.append(movie)
+    
+    #compare user movies to friend movies    
+    for my_movie in user_data["watched"]:
+        for movie in friend_movies:
+            if movie["title"] == my_movie["title"]:

```

Comment:
This second for loop is not necessary. You do not need to check title names together. You can do `if my_movie in friend_movies` to see if you should remove `my_movie` from the `friend_movies` list.